### ISSUE TITLE
Release 2024.09.06 Broke DX11 Fullscreen on Linux

### ISSUE THREAD

**Issue by AndrewJAHogue** [themrdemonized/xray-monolith#95]
Release 2024.09.06 broke fullscreen mode only in DX11, on Linux. Fails to launch, giving an `unhandled exception` error, with no actual error in the logs.
DX10 in fullscreen works, windowed and borderless in all render modes works. ONLY fullscreen in DX11 is broken.
I believe this is exclusively Linux, as there has been almost 2 months of updates and no issues from the thousands of GAMMA and EFP (windows) users. But, there has been an uptick in steam deck users getting the `unhandled exception` error.
I have confirmed with two steam deck users on the GAMMA discord that doing a clean install of the modded exes and switching to version 2024.09.04 fixes their issues.

So, all exe versions past 2024.09.04 do not work. 
I think I found the commit in 2024.09.06 that causes the issue, commit 8be09d2 specifically. I created [ my own fork ](https://github.com/Y3sMan/xray-monolith/commit/ea50259271f89063dec7c67e40fd734a1501a9d4/) reverting this commit on the latest exe (2024.10.23), compiled it, and everything seems to work again.

Info:

    Linux:
	*     Operating System: Kubuntu 24.04
	*     KDE Plasma Version: 5.27.11
	*     KDE Frameworks Version: 5.115.0
	*     Qt Version: 5.15.13
	*     Kernel Version: 6.8.0-47-generic (64-bit)
	*     Graphics Platform: X11
	*     Processors: 24 × 13th Gen Intel® Core™ i7-13700K
	*     Memory: 31.1 GiB of RAM
	*     Graphics Processor: NVIDIA GeForce RTX 2080 Ti/PCIe/SSE2
	*     Manufacturer: ASRock
	*     Product Name: Z690 Pro RS
    
    Wine:
    * Tested using various Wine and proton versions, in flatpak Lutris and system lutris, in flatpak steam and system steam. 
    * Currently use wine-ge-8-26-x86_64
    * dxvk v2.4.1
    * vkd3d v2.13
    * dxvk-nvapi v0.7.1
   
    [xray_allinux.log](https://github.com/user-attachments/files/17509465/xray_allinux.log)

**Comment by bananoviiblinchik** [comment]
The issue still persists on version 2024.11.10

**Comment by AFWL** [comment]
Can confirm, this issue is also affecting me on my Arch system with the same wine version.

**Comment by am-am9** [comment]
no, its not broke, MAYBE you forget to add sapi, here i marked it blue, i can start it from MO2 and chose DX 11, and it's works (yep i don't try it with some mod yet but it works with modded exes
![Screenshot from 2025-01-12 15-24-44](https://github.com/user-attachments/assets/75d9ea76-84f2-4872-8107-0bb05640ec8a)

edit: i maybe kinda late but maybe i can give little help :)

**Comment by AndrewJAHogue** [comment]
> no, its not broke, MAYBE you forget to add sapi, here i marked it blue, i can start it from MO2 and chose DX 11, and it's works (yep i don't try it with some mod yet but it works with modded exes ![Screenshot from 2025-01-12 15-24-44](https://private-user-images.githubusercontent.com/194527000/402325779-75d9ea76-84f2-4872-8107-0bb05640ec8a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY4ODYxMjEsIm5iZiI6MTczNjg4NTgyMSwicGF0aCI6Ii8xOTQ1MjcwMDAvNDAyMzI1Nzc5LTc1ZDllYTc2LTg0ZjItNDg3Mi04MTA3LTBiYjA1NjQwZWM4YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExNFQyMDE3MDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNzMxNWZjMWZmNjFlMjZhZDExNzUwMDRlODA1ZGU3NjQyYmFjMzZiNDFmNmI5MzZlMGMzNmNmZTg3YzI4MzQ0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.IyTWuM5eIYq9VYa069NEzR-L--q5-X4moEHfqWcWw_4)
> 
> edit: i maybe kinda late but maybe i can give little help :)

I've tried adding the SAPI override, launching vanilla 1.5.3 with exes version 2025.1.13, and still the same errors, "Unhandled exception occurred." No information in the logs, just an empty stack trace. I've tried both AVX and non-avx to the same result.

**Comment by am-am9** [comment]
Try version 2024.12.11, it's works in for me;
No, i don't download and use OpenAl runtime.

**Comment by AndrewJAHogue** [comment]
> Try version 2024.12.11, it's works in for me; No, i don't download and use OpenAl runtime.

OpenAI runtime was not mentioned at any point? 
Anyways, 2024.12.11 doesn't work, still, as neither do the latest releases due to the same issues I've been espousing since the beginning. The reversal of recent commits to the DX10HW.cpp file, up to commit 8be09d2, fixes dx11 fullscreen on linux; I've been compiling my own versions, forked from the latest exe releases, performing this change myself, and everything has been fine.

**Comment by Diver27** [comment]
Confirmed the same issue in version 2025.2.7, running on Arch Linux/KDE Wayland with wine-ge 8.26. Also confirmed that using borderless windowed is a good workaround.

**Comment by am-am9** [comment]
Nah sorry man, it's work on me, this is weird suggest from me, but, have you ever tried download the font for your wineprefix (download all the font if you can)? Maybe it can fix it.

**Comment by themrdemonized** [comment]
[AnomalyDX11AVX.zip](https://github.com/user-attachments/files/19265312/AnomalyDX11AVX.zip)
Try this exe with `--dxgi-old` argument. You can add it into `commandline.txt` file or via making a shotcut to exe and adding argument there

**Comment by AndrewJAHogue** [comment]
That works well. Thank you. Can you merge this into the main, and mention the fix for Linux users somewhere?

**Comment by themrdemonized** [comment]
Yea, the next version will have this fix
